apps/src/music/progress/MusicValidator.ts

```diff
@@ -30,16 +31,41 @@ export default class MusicValidator extends Validator {
   constructor(
     private readonly getIsPlaying: () => boolean,
     private readonly getPlaybackEvents: () => PlaybackEvent[],
+    private readonly getExemplarPlaybackEvents: () => PlaybackEvent[],
     private readonly getValidationTimeout: () => number,
     private readonly player: MusicPlayer,
     private readonly getPlayingTriggers: () => PlayingTrigger[],
+    private readonly getExemplarSettings: () => ExemplarSettings | undefined,
     private readonly conditionsChecker: ConditionsChecker = new ConditionsChecker(
       Object.values(MusicConditions).map(condition => condition.name)
     )
   ) {
     super();
   }
 
+  shouldValidateWithExemplar(): boolean {
+    const isEditingExemplar = getAppOptionsEditingExemplar();
+    return (
+      !isEditingExemplar &&
+      (this.getExemplarSettings()?.validationEnabled ?? false)
+    );
+  }
+
+  getExemplarValidationResults(): {satisfied: boolean; message: string} {
+    if (!this.shouldValidateWithExemplar()) {
+      throw new Error(
+        'Exemplar validation is disabled; cannot retrieve exemplar validation results.'
+      );
+    }
+    const satisfied = this.validatePlaybackEventsEquivalent();
+    return {
+      satisfied,
+      message: satisfied
+        ? this.getExemplarSettings()!.validationSuccessMessage
+        : this.getExemplarSettings()!.validationFailureMessage,
```

Comment:
These messages are also something the ProgressManager could potentially grab directly since they exist on the level properties object, rather than having the lab-specific validator pass them back up.

Reply:
Updated. A previous reply about creating a new setter in the `ProgressManager` class applies here as well.